specifyweb/frontend/js_src/lib/components/Merging/autoMerge.ts

```diff
@@ -73,26 +74,60 @@ const sortResources = (
 function mergeField(
   field: LiteralField | Relationship,
   resources: RA<SerializedResource<AnySchema>>,
-  cautious: boolean
+  cautious: boolean,
+  targetId?: number
 ) {
-  const values = resources.map((resource) => resource[field.name]);
+  const parentChildValues = resources.map((resource) => [
+    resource.id,
+    resource[field.name],
+  ]);
+  const values = parentChildValues.map(([_, child]) => child);
   const nonFalsyValues = f.unique(values.filter(Boolean));
   const firstValue = nonFalsyValues[0] ?? values[0];
   if (field.isRelationship)
     if (field.isDependent())
       if (relationshipIsToMany(field)) {
-        const records = nonFalsyValues as unknown as RA<
-          RA<SerializedResource<AnySchema>>
-        >;
         // Remove duplicates
-        return f
-          .unique(
-            records
-              .flat()
-              .map((value) => resourceToGeneric(value, false))
-              .map((resource) => JSON.stringify(resource))
-          )
-          .map((resource) => JSON.parse(resource));
+        const uniqueDependentsCombined = f.unique(
+          parentChildValues
+            .flatMap(([_, childResources]) =>
+              (
+                childResources as unknown as RA<SerializedResource<AnySchema>>
+              ).map((child) => resourceToGeneric(child, false))
+            )
+            .map((resource) => JSON.stringify(resource))
+        );
+
+        const resourcesToReturn = uniqueDependentsCombined.map((resource) => {
+          const parentResources =
+            /*
+             * Don't preserve dependents if targetId is not defined. This will happen if autoMerge gets called recursively for -to-one
+             * resources, but those resources also have dependent resources.
+             * TODO: Handle this case better
+             */
+            targetId === undefined
+              ? undefined
+              : parentChildValues
+                  .find(([parentId]) => parentId === targetId)
+                  ?.at(1);
```

Comment:
This value is being calculated for every stringified resource in the `uniqueDependentsCombined` Array, but it does not depend on `resource`. 

Can this be moved before `resourcesToReturn` is initialized, and thus before the `uniqueDependentsCombined.map((resource) => {...})`?

Reply:
Good catch!